### PR TITLE
Lesson 4 Assignments

### PR DESCRIPTION
**Pull request by bcamp3**
- mailroom_pt2.py : Updated mail room cmd line script using dictionary as database
- trigrams.py : Trigrams script tested with short phrase, sherlock_small.txt, and sherlock.txt examples

### CODE REVIEW

File: students/bcamp3/lesson04/mailroom_pt2.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                 '2': create_report,
+                 '3': send_letters_to_all_donors,
+                 '4': quit_msg,
+                 }
+
+# main_dispatch.setdefault('0', 'nothing')
+
+if __name__ == "__main__":
+    """Execute the main menu prompt."""
+    menu_selection(main_prompt, main_dispatch)
```
Comment (charlesrobison) [inline students/bcamp3/lesson04/mailroom_pt2.py:133]: Looks good!

File: students/bcamp3/lesson04/trigrams.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        sys.exit(1)
+    in_data = read_in_data(filename)
+    words = get_words(in_data)
+    trigrams = build_trigrams(words)
+    # for n, item in enumerate(trigrams.items()):
+    #     if n < 30:
+    #         print(item)
+    #     else:
+    #         break
+    print(use_trigrams(trigrams))
```
Comment (charlesrobison) [inline students/bcamp3/lesson04/trigrams.py:101]: Getting an error when I run this with sherlock_small.txt - what file did you have success with?

File: students/bcamp3/lesson05/mailroom_pt3.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+if __name__ == "__main__":
+    """Execute the main menu prompt."""
+    while True:
+        try:
+            menu_selection(main_prompt, main_dispatch)
+        except KeyboardInterrupt:
+            print("\nPlease use the menu option '4' to Quit.")
+        else:
+            break
```
Comment (charlesrobison) [inline students/bcamp3/lesson05/mailroom_pt3.py:161]: Works well - good exceptions

**Comment by bcamp3** [comment]
I realize I added features to trigrams.py that limited it to processing
only the full book "sherlock.txt".  It happens in "read_in_data()" where I
look for the beginning and the end of the body of the book.  If those flags
aren't found the data/text isn't read-in.  I've added a check to see if the
first line in the file is the Project Gutenberg book, if it is , process as
usual, if the book header isn't found it processes all lines.  If you would
like I can commit that version.

File: students/bcamp3/lesson04/trigrams.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    """Read text file."""
+    body = False
+    data = ''
+    with open(filename, 'r') as file:
+        for line in file:
+            if line[:9] == '*** START':
+                body = True
+            if line[:7] == '*** END':
+                body = False
+            if body and not line.isupper():
```
Comment (bcamp3) [inline students/bcamp3/lesson04/trigrams.py:24]: At this line the 'body' of the book needs to be identified for data to be read-in.  If 'body' is not found, data is not read.

**Pull request merged**